### ISSUE TITLE
Loss function

### ISSUE THREAD

**Issue by Linda230** [uds-lsv/MCSE#3]
"       for step in range(num_update_steps_per_epoch):
            model.train()

            if step % paired_sample_step == 0:
                batch = next(pair_loader)
                for key in batch.keys():
                    batch[key] = batch[key].to('cuda')
            else:
                batch = next(textonly_loader)
                for key in batch.keys():
                    batch[key] = batch[key].to('cuda')

            if step % paired_sample_step == 0 and args.framework.lower() == 'mcse':
                intra_loss, inter_loss =  model.compute_loss(batch, cal_inter=True)
                loss = intra_loss + args.lbd * inter_loss
            else:
                loss = model.compute_loss(batch, cal_inter=False)"
I am more interested in the computation of loss function, the paper mentions the final loss should be L_wiki + \lambda L_flickr_pair, but here the loss function either returns L_wiki, or L_flickr_text + \lambda L_flickr_pair. if my understanding is wrong, could you please help me with the meaning of the two loss functions in the code, thanks.

**Comment by mrzhang11** [comment]
Hey, thank you for your interest! 

>  the paper mentions the final loss should be L_wiki + \lambda L_flickr_pair

Actually the loss function is L_text + L_multimodal, so,
  - For wiki data, the loss only has a single item (i.e., L_text).
  - For flicker with paired data, the loss is a combination of two terms.  

In the code implementation, we optimize them alternately. Please let me know if you have any further questions!

**Comment by Linda230** [comment]
Thank you very much for your reply!
You mean for flicker with paired data, the loss is a combination of two terms. I think you should refers to this loss function:
if step % paired_sample_step == 0 and args.framework.lower() == 'mcse':
            intra_loss, inter_loss =  model.compute_loss(batch, cal_inter=True)
            loss = intra_loss + args.lbd * inter_loss
In these codes, I think batch data only include the paired text-image data because it is obtained from the following code
if step % paired_sample_step == 0:
            batch = next(pair_loader)
            for key in batch.keys():
                batch[key] = batch[key].to('cuda')
So I do not think "For flicker with paired data, the loss is a combination of two terms."
If I've misunderstood, please point it out to me, thanks!
I am looking forward to your reply.

**Comment by mrzhang11** [comment]
Hey, 

>  I do not think "For flicker with paired data, the loss is a combination of two terms."

Why? As you mentioned,

> In these codes, I think batch data only include the paired text-image data because it is obtained from the following code

So you can compute the L_text by using the captions only and compute the L_multimodal by using both the captions and images.

**Comment by Linda230** [comment]
Thank you.
For flicker with paired data, the loss is a combination of two terms. 
Yeah, it should be  L_text  + L_multimodal, but L_text is the loss of captions in the multimodal dataset, which is not consistent with the final loss function in your paper L_wiki_text + L_multimodal.

I am wondering how to calculate the loss of L_wiki_text + L_multimodal?

**Comment by mrzhang11** [comment]
Hi, there is no  L_wiki_text loss in the paper == 

The first loss item is called "textual objective" (not limited to wiki data), and the second item is "multimodal objective".

**Comment by Linda230** [comment]
OK. Thank you very much!
